File: Array/ThreeSum.py
from nose.tools import assert_equal

def threeSum(arr):
	arr = sorted(arr)
	l = len(arr)
	for i in range(l):
		j = i+1
		k = l-1
		while j < k:
			total = arr[i] + arr[j] + arr[k]
			if total == 0:
				return True
			elif total < 0:
				j += 1
			else:
				k -= 1
	return False

class ThreeSum(object):
	def test(self, func):
		assert_equal(func([9, 2, 1, 7, -5, -2, 8]), True)
		assert_equal(func([5, 2, 1, 0, 4]), False)
		assert_equal(func([5, 0, 6, 1, -6, 2]), True)
		print("TESTS PASSED")

t = ThreeSum()
t.test(threeSum)

File: Array/ThreeSum.js
function threeSum(arr){
	arr.sort();
	l = arr.length;
	for(var i=0; i<l; i++){
		j = i+1;
		k = l;
		while(j < k){
			total = arr[i]+arr[j]+arr[k];
			if(total === 0){
				return true;
			} else if(total < 0){
				j++;
			} else{
				k--;
			}
		}
	}
	return false;
}

console.log(threeSum([5, 2, 1, 7, -5, 0]));
console.log(threeSum([5, 2, 1, 7, 0]));